File: Python/509. Fibonacci Number.py
class Solution:
    def fib(self, n: int) -> int:
#         def fib(n):
#             if n == 0 or n == 1: return n
#             return fib(n - 1) + fib(n - 2)
#         return fib(n)

        fib = [0, 1]
        for i in range(1, n + 1):
            fib.append(fib[i] + fib[i - 1])
        return fib[n]

File: PHP/509. Fibonacci Number.php
<?php
class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function fib($n) {
        $fib = array(0, 1);
        for($i = 2; $i <= $n; $i++){
            $fib[$i] = $fib[$i - 1] + $fib[$i - 2];
        }
        return $fib[$n];
    }
}
